src/main/java/com/example/decafe/PressedButtons.java

```diff
@@ -0,0 +1,53 @@
+package com.example.decafe;
+
+import javafx.animation.AnimationTimer;
+import javafx.beans.binding.BooleanBinding;
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.fxml.FXML;
+import javafx.scene.input.KeyEvent;
+
+public class PressedButtons {
+  public BooleanProperty w = new SimpleBooleanProperty();
+  public BooleanProperty a = new SimpleBooleanProperty();
+  public BooleanProperty s = new SimpleBooleanProperty();
+  public BooleanProperty d = new SimpleBooleanProperty();
+
+  private BooleanBinding any = w.or(a).or(s).or(d);
+
+  public void handleButtonPress(AnimationTimer timer) {
+    any.addListener((observableValue, aBoolean, t1) -> { // if any key from the four keys is pressed
+      if (!aBoolean) {
+          timer.start();
+      } else {
+          timer.stop();
+      }
+    });
+  }
+
+  // key events if wasd-keys are pressed
+  @FXML
+  public void keyPressed(KeyEvent event) {
+      switch (event.getCode()) {
```

Comment:
What for actually? Won't it be a line that does nothing?
Or is it a convention to always add it?

Reply:
You don't cover all possible cases, it makes it easier to find bugs.

Reply:
That's truth, but in this when a player presses other button than w,a,s,d we do not want to react at all.
For a second I decided to throw an `IllegalStateException`, but it does not make much sense, because it's actaully okay that the user clicks a button that does not bother the application,  does it? 

How would you handle such a case?

Reply:
I would still add such a case, even if it's empty.